src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -53,9 +54,16 @@ public IDisposable Start(IFeatureCollection serverFeatures, Func<IFeatureCollect
             try
             {
                 var information = (KestrelServerInformation)serverFeatures.Get<IKestrelServerInformation>();
-                var engine = new KestrelEngine(_libraryManager, new ServiceContext { AppShutdown = _appShutdownService, Log = new KestrelTrace(_logger) });
+                var dateHeaderValueManager = new DateHeaderValueManager();
```

Comment:
This is a little messy, as whoever creates the `DateHeaderManager` should be disposing it, and `ServiceContext` isn't disposable, and everything else derives from it, and you don't want them to destroy the default instance of `DateHeaderManager` in their base. We might have to fix this more generically later for things like `MemoryPool` as well that hang off of `ServiceContext`.